Update Yasson library to support java 17 records

### Description

Please update the [Yasson depedency to at least 2.0.3](https://github.com/eclipse-ee4j/yasson/releases/tag/2.0.3) for proper record support. Currently (2.5.2) only 1.0.9 is shipped.

Overriding the dependency to force 2.0.3 breaks the quarkus compilation:

```
Caused by: java.lang.ClassNotFoundException: jakarta.json.bind.spi.JsonbProvider
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:455)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:405)
```

### Implementation ideas

_No response_

I am having some issue with running all tests locally (even before changing the BOM), which are probably caused by my local setup (had the same issue with Yasson due to localisation). 
`[ERROR]   ConfigPropertiesTest.configProperties:99 expected: <empty> but was: <localhost.localdomain>` 
Hence I pushed the PR without verifying locally and relying on the workflow. 
@gsmet somebody needs to approve the workflow to start the ci.

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -155,7 +155,7 @@
         <awaitility.version>4.1.1</awaitility.version>
         <jboss-logmanager.version>1.0.9</jboss-logmanager.version>
         <flyway.version>8.4.1</flyway.version>
-        <yasson.version>1.0.10</yasson.version>
+        <yasson.version>1.0.11</yasson.version>
         <liquibase.version>4.6.2</liquibase.version>
         <liquibase-mongodb.version>4.6.2</liquibase-mongodb.version>
         <snakeyaml.version>1.30</snakeyaml.version>

--- a/independent-projects/resteasy-reactive/pom.xml
+++ b/independent-projects/resteasy-reactive/pom.xml
@@ -59,7 +59,7 @@
         <jackson.version>2.13.1</jackson.version>
         <smallrye-stork.version>1.0.0.Alpha7</smallrye-stork.version>
         <jakarta.validation-api.version>2.0.2</jakarta.validation-api.version>
-        <yasson.version>2.0.2</yasson.version>
+        <yasson.version>1.0.11</yasson.version>
         <jakarta.json.bind-api.version>1.0.2</jakarta.json.bind-api.version>
         <awaitility.version>4.1.1</awaitility.version>
         <smallrye-mutiny-vertx-core.version>2.18.0</smallrye-mutiny-vertx-core.version>

